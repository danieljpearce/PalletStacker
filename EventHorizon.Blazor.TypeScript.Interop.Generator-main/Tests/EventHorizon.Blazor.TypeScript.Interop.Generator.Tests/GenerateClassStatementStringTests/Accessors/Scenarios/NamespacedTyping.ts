declare module Accessors.Scenarios {
    export class ExampleClass {
        get namespacedTyping(): BABYLON.Mesh.AbstractMesh;
        get notNamespacedTyping(): AbstractMesh;
    }
}