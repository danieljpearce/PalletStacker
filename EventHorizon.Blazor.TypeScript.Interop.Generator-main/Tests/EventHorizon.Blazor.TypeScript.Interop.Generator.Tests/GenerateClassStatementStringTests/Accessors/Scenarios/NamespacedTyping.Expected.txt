/// Generated - Do Not Edit
namespace Accessors.Scenarios
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : CachedEntityObject
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors
        private AbstractMesh __namespacedTyping;
        public AbstractMesh namespacedTyping
        {
            get
            {
            if(__namespacedTyping == null)
            {
                __namespacedTyping = EventHorizonBlazorInterop.GetClass<AbstractMesh>(
                    this.___guid,
                    "namespacedTyping",
                    (entity) =>
                    {
                        return new AbstractMesh() { ___guid = entity.___guid };
                    }
                );
            }
            return __namespacedTyping;
            }
        }

        private AbstractMesh __notNamespacedTyping;
        public AbstractMesh notNamespacedTyping
        {
            get
            {
            if(__notNamespacedTyping == null)
            {
                __notNamespacedTyping = EventHorizonBlazorInterop.GetClass<AbstractMesh>(
                    this.___guid,
                    "notNamespacedTyping",
                    (entity) =>
                    {
                        return new AbstractMesh() { ___guid = entity.___guid };
                    }
                );
            }
            return __notNamespacedTyping;
            }
        }
        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods

        #endregion
    }
}